bundles/org.openhab.binding.novafinedust/src/main/java/org/openhab/binding/novafinedust/internal/sds011protocol/SDS011Communicator.java

```diff
@@ -0,0 +1,287 @@
+     *
+     * @throws IOException
+     */
+    public void requestSensorData() throws IOException {
+        CommandMessage m = new CommandMessage(Command.REQUEST_DATA, new byte[] {});
+        byte[] data = m.getBytes();
+        if (logger.isDebugEnabled()) {
+            logger.debug("Requesting sensor data, will send: {}", Helper.toHexString(data));
+        }
+        outputStream.write(data, 0, data.length);
+    }
+
+    private @Nullable SensorReply readReply() throws IOException {
+        byte[] readBuffer = new byte[Constants.REPLY_LENGTH];
+
+        int b = -1;
+        if (inputStream.available() > 0) {
+            while ((b = inputStream.read()) != Constants.MESSAGE_START_AS_INT) {
+                logger.debug("Trying to find first reply byte now...");
+            }
+            readBuffer[0] = (byte) b;
+            int remainingBytesRead = inputStream.read(readBuffer, 1, Constants.REPLY_LENGTH - 1);
+            if (logger.isDebugEnabled()) {
+                logger.debug("Read remaining bytes: {}, full reply={}", remainingBytesRead,
+                        Helper.toHexString(readBuffer));
+            }
+            return ReplyFactory.create(readBuffer);
+        }
+        return null;
+    }
+
+    /**
+     * Data from the device is arriving and will be parsed accordingly
+     */
+    @Override
+    public void serialEvent(SerialPortEvent event) {
+        if (event.getEventType() == SerialPortEvent.DATA_AVAILABLE) {
+            // we get here if data has been received
+            SensorReply reply = null;
+            try {
+                reply = readReply();
+                logger.debug("Got data from sensor: {}", reply);
+            } catch (IOException e) {
+                logger.error("Could not read available data from the serial port", e);
+            }
+            if (reply instanceof SensorMeasuredDataReply) {
+                SensorMeasuredDataReply sensorData = (SensorMeasuredDataReply) reply;
+                if (sensorData.isValidData()) {
+                    thingHandler.updateChannels(sensorData);
+                }
+            }
+        }
+    }
+
+    /**
+     * Shutdown the communication, i.e. send the device to sleep and close the serial port
+     */
+    public void dispose() {
+        if (serialPort != null) {
+            try {
+                // send the device to sleep to preserve power and extend the lifetime of the sensor
+                sendSleep(true);
+            } catch (IOException e) {
+                // ignore because we are shutting down anyway
+                logger.debug("Exception while disposing communicator (will ignore it)", e);
+            } finally {
+                serialPort.removeEventListener();
+                serialPort.close();
+                serialPort = null;
+            }
+        }
+        IOUtils.closeQuietly(inputStream);
+        IOUtils.closeQuietly(outputStream);
```

Comment:
Since we are going to remove the Apache Commons dependencies, could you replace the lines with standard Java code? See #7722